### ISSUE TITLE
YAML template processor strips ${{property}} placeholders to bare $ in violation messages

### ISSUE THREAD

**Issue by tsanders-rh** [konveyor/analyzer-lsp#1051]
## Issue Description

The Konveyor analyzer is incorrectly processing template placeholders in rule messages, stripping `${{property.name}}` syntax down to bare `$`, which generates invalid XML/code examples in violation messages.

## Evidence

**Source Rule File**: `default/generated/quarkus/216-javaee-pom-to-quarkus.windup.yaml`

The source YAML has **CORRECT** syntax using `${{property.name}}`:

```yaml
message: "Use the Quarkus BOM...
  <groupId>${{quarkus.platform.group-id}}</groupId>
  <artifactId>${{quarkus.platform.artifact-id}}</artifactId>
  <version>${{quarkus.platform.version}}</version>
```

**Generated Violation Message** (in analysis output) has **INVALID** syntax with bare `$`:

```xml
<groupId>$</groupId>
<artifactId>$</artifactId>
<version>$</version>
```

## Expected Behavior

Template placeholders should be converted to valid Maven property syntax:
- `${{property.name}}` → `${property.name}`

## Actual Behavior

Template placeholders are stripped to invalid bare `$`:
- `${{property.name}}` → `$`

## Impact

1. **Invalid XML/code**: Bare `$` is not valid XML or Maven syntax
2. **AI agent confusion**: AI-assisted migration tools either:
   - Copy the invalid `$` literally (breaks builds)
   - Try to "fill in" the placeholder incorrectly (hallucinations)
3. **Manual migration difficulty**: Developers see invalid examples that must be mentally corrected

## Examples of Affected Rules

All rules in `216-javaee-pom-to-quarkus.windup.yaml` are affected:

| Rule ID | Lines | Invalid Output |
|---------|-------|----------------|
| javaee-pom-to-quarkus-00010 | 38-40 | `<groupId>$</groupId>` |
| javaee-pom-to-quarkus-00020 | 66-67 | `<version>$</version>` |
| javaee-pom-to-quarkus-00030 | 92 | `<version>$</version>` |
| javaee-pom-to-quarkus-00040 | 118, 120 | `<version>$</version>`, `<maven.home>$</maven.home>` |
| javaee-pom-to-quarkus-00050 | 145, 147-149 | `<native.image.path>$/$-runner</native.image.path>` |

## Root Cause Analysis

The YAML template processor appears to be:
1. Recognizing `${{...}}` as a template variable
2. But stripping the content entirely instead of converting to target syntax
3. Likely a regex/substitution bug in the message rendering logic

## Suggested Fix

Update the YAML template processor to:
- Preserve property names: `${{property.name}}` → `${property.name}`
- Or if properties should be evaluated, populate them with actual values instead of `$`
- At minimum, document the expected syntax for rule authors

## Reproduction

1. Analyze any JavaEE project with Konveyor targeting Quarkus
2. Examine violation messages for rules in `javaee-pom-to-quarkus`
3. Observe bare `$` placeholders in XML examples

## Related Issue

See konveyor/rulesets#328 for the complete analysis of problems in the javaee-pom-to-quarkus rules (this engine bug plus source rule file issues).

## Environment

- Konveyor version: Latest (as of 2025-12-21)
- Ruleset: `default/generated/quarkus/216-javaee-pom-to-quarkus.windup.yaml`
- Analysis tool: `kantra`
